libshpool/src/config.rs

```diff
@@ -212,6 +237,37 @@ pub struct Config {
     pub motd_args: Option<Vec<String>>,
 }
 
+impl Config {
+    // Merge with `another` Config instance, with `self` taking higher priority.
+    //
+    // Top level options with simple value are directly taken from the higher priority instance.
+    // The merge strategy for list or map are handled case by case. Please refer to each options'
+    // documentation for details.
```

Comment:
Let's also metion in this comment that this or is not communicative. Kinda sad that the stdlib has a mathy sounding function that does not obey the expected laws (maybe this is just the recovering Haskell programmer in me talking)

Reply:
Sure, done. I only learned Haskell years ago, but it never clicked for me :P